server/src/analyser.ts

```diff
@@ -525,6 +823,36 @@ export default class Analyzer {
     )
   }
 
+  public symbolAtPointFromTextPosition(
```

Comment:
This method is like `wordAtPointFromTextPosition` above it, but returns only specific symbols (i.e., variables and functions), doesn't necessarily have to be a part of a declaration. I created this to make sure what gets captured from a given position is something that can actually be renamed rather than a flag or other `word`s that can't.

Since you brought it up, I'm actually thinking of changing the signature of this to return a `SymbolKind` to stick with LSP semantics rather than making my own with the `type` property. I think doing that would help make  what is being returned more explicit.